cpp/oneapi/dal/algo/svm/test/serialization.cpp

```diff
@@ -0,0 +1,238 @@
+
+        return { homogen_table::wrap(x_train, row_count, feature_count),
+                 homogen_table::wrap(y_train, row_count, 1) };
+    }
+
+    template <typename T = task_t, detail::enable_if_classification_t<T>* = nullptr>
+    table get_test_data() {
+        constexpr std::int64_t row_count = 9;
+        constexpr std::int64_t feature_count = 3;
+        static const float_t x_test[] = {
+            0.4213,  1.5162,  0.1232, //
+            -0.9252, -0.514,  -0.4708, //
+            0.87,    0.8338,  3.4986, //
+            1.1788,  -0.5077, 4.5583, //
+            -1.4726, 2.2684,  -1.8172, //
+            -0.0091, -1.0776, 1.5261, //
+            -0.5016, -2.9235, 0.3886, //
+            0.4889,  -0.2868, 0.6805, //
+            -2.0861, 0.9992,  -1.9605, //
+        };
+        return homogen_table::wrap(x_test, row_count, feature_count);
+    }
+
+    template <typename T = task_t, detail::enable_if_classification_t<T>* = nullptr>
+    descriptor_t get_descriptor() {
+        ONEDAL_ASSERT(class_count_ > 0);
+        return descriptor_t{}.set_class_count(class_count_);
+    }
+
+    template <typename T = task_t, detail::enable_if_regression_t<T>* = nullptr>
+    descriptor_t get_descriptor() {
+        return descriptor_t{};
+    }
+
+    model<task_t> train_model() {
+        const auto [x_train, y_train] = this->get_train_data();
+        return this->train(this->get_descriptor(), x_train, y_train).get_model();
+    }
+
+    infer_result<task_t> run_inference(const model<task_t>& m) {
+        return this->infer(this->get_descriptor(), m, this->get_test_data());
+    }
+
+    void compare_infer_results(const infer_result<task_t>& actual,
+                               const infer_result<task_t>& reference) {
+        INFO("compare labels") {
+            te::check_if_tables_equal<float_t>(actual.get_labels(), reference.get_labels());
+        }
+        INFO("compare decision function") {
+            te::check_if_tables_equal<float_t>(actual.get_decision_function(),
+                                               reference.get_decision_function());
+        }
+    }
+
+    void run_test() {
+        INFO("training");
+        const auto model = train_model();
+
+        INFO("serialization");
+        const auto deserialized_model = te::serialize_deserialize(model);
+
+        INFO("inference");
+        const auto expected = run_inference(model);
+        const auto actual = run_inference(deserialized_model);
+        compare_infer_results(actual, expected);
+    }
+
+private:
+    std::int64_t class_count_ = -1;
+};
+
+using svm_cls_types = COMBINE_TYPES((float, double),
+                                    (svm::method::thunder, svm::method::smo),
+                                    (svm::task::classification));
+
+using svm_reg_types = COMBINE_TYPES((float, double),
+                                    (svm::method::thunder, svm::method::smo),
```

Comment:
for regression not support SMO method
```suggestion
                                    (svm::method::thunder),
```